syntax = "proto3";

// The Account service definition
service Reply {
  // Sends a confirmation
  rpc Creation (CreationRequest) returns (CreationReply) {}
  rpc Login (LoginRequest) returns (LoginReply) {}
  rpc Delete (DeleteRequest) returns (DeleteReply) {}
  rpc Update (UpdateRequest) returns (UpdateReply) {}
}


// Request message containing the 'new' account credentials
message CreationRequest {
  string name = 1;
  string password = 2;
}

// Response Message containing the Account Creation status
message CreationReply {
  string message = 1;
}

// Request message containing login credentials
message LoginRequest {
  string name = 1;
  string password = 2;
  string message = 3;
}

// Response message containing the status of a Login
message LoginReply {
  string message = 1;
  string token = 2;
}

// Request message containing Deletion credentials
message DeleteRequest {
  string name = 1;
  string password = 2;
  string authentication = 3;
  string message = 4;
}

// Response message containing the status of a Login
message DeleteReply {
  string message = 1;
}

// Request message containing Deletion credentials
message UpdateRequest {
  string name = 1;
  string password = 2;
  string newPassword = 3;
  string authentication = 4;
  string message = 5;
}

// Response message containing the status of a Login
message UpdateReply {
  string message = 1;
}
